datasets/style_change_detection/style_change_detection.py

```diff
@@ -0,0 +1,127 @@
+  title={Shared Tasks on Authorship Analysis at PAN 2020},
+  author={Bevendorff, Janek and Ghanem, Bilal and Giachanou, Anastasia and Kestemont, Mike and Manjavacas, Enrique and Potthast, Martin and Rangel, Francisco and Rosso, Paolo and Specht, G{\"u}nther and Stamatatos, Efstathios and others},
+  booktitle={European Conference on Information Retrieval},
+  pages={508--516},
+  year={2020},
+  organization={Springer}
+}
+"""
+
+_DESCRIPTION = """\
+The goal of the style change detection task is to identify text positions within a given multi-author document at which the author switches. Detecting these positions is a crucial part of the authorship identification process, and for multi-author document analysis in general.
+
+Access to the dataset needs to be requested from zenodo.
+"""
+
+
+class StyleChangeDetection(nlp.GeneratorBasedBuilder):
+    """Style Change Detection Dataset from PAN20"""
+
+    VERSION = nlp.Version("1.0.0")
+    BUILDER_CONFIGS = [
+        nlp.BuilderConfig(
+            name="narrow",
+            version=nlp.Version("1.0.0", "Version 1"),
+            description="The narrow subset contains texts from a relatively narrow set of subjects matters (all related to technology).",
+        ),
+        nlp.BuilderConfig(
+            name="wide",
+            version=nlp.Version("1.0.0", "Version 1"),
+            description="The wide subset adds additional subject areas (travel, philosophy, economics, history, etc.).",
+        ),
+    ]
+
+    @property
+    def manual_download_instructions(self):
+        return """\
+  You should download the dataset from https://zenodo.org/record/3660984
+  The dataset needs requesting.
+
+  Download each file, extract it and place in a dir of your choice,
+  which will be used as a manual_dir, e.g. `~/.manual_dirs/style_change_detection`
+  Style Change Detection can then be loaded via:
+  `nlp.load_dataset("style_change_detection", data_dir="~/.manual_dirs/style_change_detection")`.
+  """
+
+    def _info(self):
+        features = {
+            "id": nlp.Value("string"),
+            "text": nlp.Value("string"),
+            "authors": nlp.Value("int32"),
+            "structure": nlp.features.Sequence(nlp.Value("string")),
+            "site": nlp.Value("string"),
+            "multi-author": nlp.Value("bool"),
+            "changes": nlp.features.Sequence(nlp.Value("bool")),
+        }
+
+        return nlp.DatasetInfo(
+            description=_DESCRIPTION,
+            features=nlp.Features(features),
+            homepage="https://pan.webis.de/clef20/pan20-web/style-change-detection.html",
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = os.path.abspath(os.path.expanduser(dl_manager.manual_dir))
+        train_dir = os.path.join(data_dir, "train", "dataset-" + self.config.name)
+        val_dir = os.path.join(data_dir, "train", "dataset-" + self.config.name)
```

Comment:
We could check that the files exist here, and raise the adequate error if they don't exist, as we do in `xtreme.py` for example.

Something like:
```
            if not os.path.exists(train_dir):
                raise FileNotFoundError(
                    "{} does not exist. Make sure you insert a manual dir via `nlp.load_dataset('style_change_detection', data_dir=...)` that includes {}. Manual download instructions: {}".format(
                        train_dir, train_dir, self.manual_download_instructions
                    )
                )
```

This makes me think we should add a check like this in `hyperpartisan_news_detection.py` too

Reply:
I agree - would be nice to give a more readable error here.